example/helloWorld/src/helloworld.cpp

```diff
@@ -0,0 +1,190 @@
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * \author Erik Zenker
+ *
+ */
+// Alpaka
+#include <alpaka/alpaka.hpp>
+
+// STL
+#include <iostream>
+
+
+/**
+ * Helper functions that calculates the global thread id depending
+ * on the number of threads set by the workdiv.
+ */
+template <typename T_Acc>
+size_t globalThreadIdx(T_Acc const &acc){
+    auto threadsExtent   = alpaka::workdiv::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+    auto threadIdx       = alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+
+    /**
+     * Map the three dimensional thread index into a
+     * one dimensional thread index space.
+     */
+    auto globalThreadIdx = alpaka::core::mapIdx<1u>( threadIdx,
+						     threadsExtent );
+    return globalThreadIdx[0];
+}
+
+/**
+ * Hello World Kernel
+ *
+ * Prints "[tid] Hello World" where tid is
+ * the global thread number.
+ */
+struct HelloWorldKernel {
+    template <typename T_Acc>
+    ALPAKA_FN_ACC void operator()( T_Acc const &acc) const {
+        printf("[%u] Hello World\n", static_cast<unsigned>(globalThreadIdx(acc)));
+
+    }
+
+};
+
+int main() {
+
+    /**
+     * Define accelerator types
+     *
+     * It is possible to choose from a set of accelerators
+     * that are defined in the alpaka::acc namespace e.g.:
+     * - AccGpuCudaRt
+     * - AccCpuThreads
+     * - AccCpuFibers
+     * - AccCpuOmp2Threads
+     * - AccCpuOmp2Blocks
+     * - AccCpuOmp4
+     * - AccCpuSerial
+     *
+     * Each accelerator has strengths and weaknesses. Therefore,
+     * they need to be choosen carefully depending on the actual
+     * use case. Furthermore, some accelerators only support a
+     * particular workdiv, but workdiv can also be generated
+     * automatically.
+     *
+     */
+    using Dim     = alpaka::dim::DimInt<3>;
+    using Size    = std::size_t;
+    using Host    = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Acc     = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Stream  = alpaka::stream::StreamCpuSync;
+    using DevAcc  = alpaka::dev::Dev<Acc>;
+    using DevHost = alpaka::dev::Dev<Host>;
+    using WorkDiv = alpaka::workdiv::WorkDivMembers<Dim, Size>;
+
+
+    /**
+     * Get the first devices
+     *
+     * The accelerator only defines how something should be
+     * parallized, but a device is the real entity which will
+     * run the parallel programm. The device can be choosen
+     * by id (0 to the number of devices minus 1) or you
+     * can also retrieve all devices in a vector (getDevs()).
+     * In this example the first devices is choosen.
+     *
+     */
+    DevAcc  devAcc  (alpaka::dev::DevMan<Acc>::getDevByIdx(0));
+    DevHost devHost (alpaka::dev::DevMan<Host>::getDevByIdx(0));
+
+    
+    /**
+     * Create a stream to the accelerator device
+     *
+     * A stream can be interpreted as the work queue
+     * of a particular device. Streams are filled with
+     * execution units and alpaka takes care that these
```

Comment:
In the code the are called executor and not execution unit. Maybe it is better to use the library internal naming here and in the following comments.